packages/gatsby/src/redux/reducers/match-paths.js

```diff
@@ -0,0 +1,32 @@
+const _ = require(`lodash`)
+
+module.exports = (state = {}, action) => {
+  switch (action.type) {
+    case `DELETE_CACHE`:
+      return {}
+    case `CREATE_PAGE`: {
+      const page = action.payload
+      if (page.matchPath) {
+        const { oldPage } = action
+        const newState = { ...state }
+        if (oldPage && oldPage.matchPath !== page.matchPath) {
+          delete newState.matchMaths[oldPage.matchPath]
```

Comment:
This mutates the original `state` object, is that okay?

Reply:
Good catch. Not to mention that `matchMaths` makes no sense whatsoever 😂. Fixed in https://github.com/gatsbyjs/gatsby/pull/13012